### ISSUE TITLE
Stableswap

### ISSUE THREAD

**Issue by jonator** [osmosis-labs/osmosis-frontend#980]
Here are our final core stableswap functions:
* [Out amount given in](https://github.com/osmosis-labs/osmosis/blob/748262d76832199e9b4fda8c44114c3033ecf935/x/gamm/pool-models/stableswap/pool.go#L245)
* [In amount given out](https://github.com/osmosis-labs/osmosis/blob/748262d76832199e9b4fda8c44114c3033ecf935/x/gamm/pool-models/stableswap/pool.go#L280)
* [Core solver for both of the above (only math-y part)](https://github.com/osmosis-labs/osmosis/blob/748262d76832199e9b4fda8c44114c3033ecf935/x/gamm/pool-models/stableswap/amm.go#L239-L302)

The solver is the same for both and runs mostly boilerplate binary search on the CFMM, which is defined [here](https://github.com/osmosis-labs/osmosis/blob/748262d76832199e9b4fda8c44114c3033ecf935/x/gamm/pool-models/stableswap/amm.go#L41) (the comments about the solver function I linked in the third bullet point should explain its logic in detail) (edited) 

Basically for Out Given In, we pass in a negative yIn to the solver (i.e. the “out” amount), which gives us the correct but negative output, which we have to negate

**Comment by AlpinYukseloglu** [comment]
Unit tests:
* [Out amount given in](https://github.com/osmosis-labs/osmosis/blob/748262d76832199e9b4fda8c44114c3033ecf935/x/gamm/pool-models/stableswap/pool_test.go#L647)
* [In amount given out](https://github.com/osmosis-labs/osmosis/blob/748262d76832199e9b4fda8c44114c3033ecf935/x/gamm/pool-models/stableswap/pool_test.go#L714)
* [Core solver](https://github.com/osmosis-labs/osmosis/blob/748262d76832199e9b4fda8c44114c3033ecf935/x/gamm/pool-models/stableswap/amm_test.go#L454)

**Comment by jonator** [comment]
- [ ] Implement getNormalizedLiquidity @AlpinYukseloglu 

Stableswap implementation:
https://github.com/osmosis-labs/osmosis-frontend/blob/3a4ae28ef37505b2f4dce137fc66df24eec364d9/packages/pools/src/stable.ts#L311-L319

Weighted implementation: 
https://github.com/osmosis-labs/osmosis-frontend/blob/3a4ae28ef37505b2f4dce137fc66df24eec364d9/packages/pools/src/weighted.ts#L350-L358

**Comment by jonator** [comment]
- [ ] Iconography on pools page
- [ ] Label on pool detail page
